discovery_engine/lib/src/domain/document_manager.dart

```diff
@@ -82,6 +86,21 @@ class DocumentManager {
       throw StateError('id $id does not have active data attached');
     }
 
+    // update reacted sources repo if necessary
+    if (userReaction != UserReaction.neutral) {
+      final source = doc.resource.sourceDomain;
+      final like = userReaction == UserReaction.positive;
+      final sourceReacted = await _sourceRepo.fetchBySource(source);
+
+      if (sourceReacted == null) {
+        await _sourceRepo.save(SourceReacted(source, like));
+      } else if (sourceReacted.liked != like) {
+        await _sourceRepo.remove(source);
```

Comment:
here the save of the source is missing
```suggestion
        await _sourceRepo.remove(source);
        await _sourceRepo.save(SourceReacted(source, like));
```

Reply:
ah ok 👍 my bad!